pkg/controllers/trace_controller.go

```diff
@@ -202,10 +198,17 @@ func (r *TraceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		return ctrl.Result{}, err
 	}
 
-	// Call gadget.Operation()
+	// Call gadget operation
 	traceBeforeOperation := trace.DeepCopy()
 	patch := client.MergeFrom(traceBeforeOperation)
-	factory.LookupOrCreate(req.NamespacedName).Operation(trace, op, params)
+
+	gadgetOperation, ok := factory.Operations()[op]
+	if !ok {
+		trace.Status.OperationError = fmt.Sprintf("Unknown operation %q", op)
```

Comment:
However, gadgets.SetStatusError() changes the State to "Stopped" and it's not what happened here. If the gadget was running, it would still be running.

Reply:
Right, setting Stopped state does not make sense for Stream gadgets. But what about Status gadgets? 

Even though we are setting the error, don't you think we should clean up the Output and change the State value to avoid having the previous output and state (Potentially valid values) and an error set at the same time? Or you say that it is up to the client to ignore Status and Output if Error is set?

Reply:
I guess I can clear `trace.Status.Output` for clarity. But I am not sure about `trace.Status.State`. For example with the biolatency and network-policy gadgets , users would not know whether the monitoring is running or not if we erase `trace.Status.State`.

Reply:
Yeah, `trace.Status.State` is complicated because it is difficult to set a value that makes sense for all the gadgets. So I agree to not change it.